src/Components/Web.JS/src/Services/NavigationEnhancement.ts

```diff
@@ -239,7 +240,10 @@ export async function performEnhancedPageLoad(internalDestinationHref: string, i
           history.replaceState(null, '', response.url);
         } else {
           // For non-gets, we're still on the source page, so need to append a whole new history entry
-          history.pushState(null, '', response.url);
+          if (response.url !== previousLocation)
+          {
```

Comment:
```suggestion
          if (response.url !== previousLocation) {
```

Also, just out of curiosity, could you say why we need to capture `location.href` in the `previousLocation` variable? Would it be different if the code checked `if (response.url !== location.href)` directly here?

Reply:
I checked `if (response.url !== location.href)` and it works the same. I mistakenly assumed that the location would be changed when this check is happening. I'll change the code. Thanks!